src/osd/PG.cc

```diff
@@ -362,8 +369,15 @@ bool PG::proc_replica_info(
   might_have_unfound.insert(from);
   
   unreg_next_scrub();
-  if (info.history.merge(oinfo.history))
+  if (info.history.merge(oinfo.history)) {
     dirty_info = true;
+    if (info.history.last_epoch_clean >= info.history.same_interval_since) {
+      dout(20) << __func__ << "clearing past_intervals" << dendl;
+      past_intervals.clear();
+      dirty_big_info = true;
+    }
+  }
+
```

Comment:
this kind of code seems already exist in update_history and called by proc_master_log and proc_replica_info